sdkproject/Assets/Mapbox/Unity/Location/EditorLocationProvider.cs

```diff
@@ -21,28 +22,55 @@ public class EditorLocationProvider : AbstractEditorLocationProvider
 		string _latitudeLongitude;
 
 		/// <summary>
-		/// The mock heading value.
+		/// The transform that will be queried for location and heading data & ADDED to the mock latitude/longitude
+		/// Can be changed at runtime to simulate moving within the map.
 		/// </summary>
 		[SerializeField]
-		[Range(0, 359)]
-		float _heading;
+		Transform _targetTransform;
+
+		[SerializeField]
+		AbstractMap _map;
+
+		bool _mapInitialized;
+
+		protected override void Awake()
+		{
+			base.Awake();
+			_map.OnInitialized += Map_OnInitialized;
+		}
+
+		void Map_OnInitialized()
+		{
+			_map.OnInitialized -= Map_OnInitialized;
+			_mapInitialized = true;
+		}
 
 		Vector2d LatitudeLongitude
 		{
 			get
 			{
+				if (_mapInitialized)
+				{
+					var startingLatLong = Conversions.StringToLatLon(_latitudeLongitude);
+					var position = Conversions.GeoToWorldPosition(startingLatLong,
+																 _map.CenterMercator,
+																 _map.WorldRelativeScale).ToVector3xz();
+					position += _targetTransform.position;
```

Comment:
Add a null value check for `_transformTransform` , will be nice to be able to use EditorLocationProvider without providing a transform.